src/flood.rs

```diff
@@ -0,0 +1,78 @@
+use crate::view::layer::LayerCoords;
+use crate::view::{Snapshot, View, ViewResource};
+use rgx::color::{Rgba8};
+use rgx::kit::shape2d::{Fill, Rotation, Shape, Stroke};
+use rgx::kit::ZDepth;
+use rgx::rect::Rect;
+
+pub fn flood_fill(
+    view: &View<ViewResource>,
+    starting_point: LayerCoords<u32>,
+    replacement_color: Rgba8,
+) -> Option<Vec<Shape>> {
+    let (snapshot, pixels) = view.current_snapshot(view.active_layer_id)?;
+    let mut canvas = pixels.clone().into_rgba8();
```

Comment:
Instead of cloning the canvas which could be expensive, could you just call `pixels.get()` directly in `get_color`? It returns an `Rgba8`.

Reply:
We need some sort of mutable storage to keep us from re-filling pixels. I was worried about modifying the snapshot's pixels directly (as I'm not sure if those are safe to change), thus the clone.

Reply:
Yup, this makes sense now, given the other comment.